src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,268 @@
+        } else if (isAllNonNullUsed(tree)) {
+            tree.arguments().forEach(contentResource -> contentResources.replace(contentResource.toString(), true));
+        } else if (insideIfStatement && isThisIsNullMethod(tree)) {
+            insideEqualNullCheck = true;
+        }
+    }
+
+    private void chainedMethodsNullCheck(MethodInvocationTree tree) {
+        if (tree.methodSelect() instanceof MemberSelectExpressionTree) {
+            MemberSelectExpressionTree method = (MemberSelectExpressionTree) tree.methodSelect();
+            if (method.expression() instanceof MethodInvocationTree) {
+                MethodInvocationTree invocation = (MethodInvocationTree) method.expression();
+                if (!invocation.symbol().isUnknown() &&
+                    isPage(invocation.symbol().owner().type().fullyQualifiedName()) &&
+                    isGetContentResourceUsedOnPage(invocation) &&
+                    !returnOccurredInsideEqualNullCheck) {
+                    context.reportIssue(this, tree, RULE_MESSAGE);
+                }
+            }
+        }
+    }
+
+    private boolean isResource(AssignmentExpressionTree tree) {
+        return Constants.RESOURCE_TYPE.equals(tree.variable().symbolType().fullyQualifiedName());
+    }
+
+    private boolean isResourceInitialized(VariableTree tree) {
+        return tree.initializer() != null && tree.initializer().kind() != Kind.NULL_LITERAL;
+    }
+
+    private boolean isGetContentResourceUsedOnPage(MethodInvocationTree tree) {
+        return GET_CONTENT_RESOURCE_METHOD.equals(tree.methodSelect().lastToken().text());
+    }
+
+    private boolean isGetContentResourceUsedOnPage(VariableTree tree) {
+        return tree.initializer() instanceof MethodInvocationTree &&
+            !((MethodInvocationTree) tree.initializer()).symbol().isUnknown() &&
+            isPage(((MethodInvocationTree) tree.initializer()).symbol().owner().type()
+                .fullyQualifiedName()) &&
+            GET_CONTENT_RESOURCE_METHOD.equals(((MethodInvocationTree) tree.initializer()).symbol().name());
+    }
+
+    private boolean isGetContentResourceUsedOnResource(AssignmentExpressionTree tree) {
+        return isResource(tree) &&
+            tree.expression() instanceof MethodInvocationTree &&
+            GET_CONTENT_RESOURCE_METHOD.equals(((MethodInvocationTree) tree.expression()).symbol().name());
+    }
+
+    private boolean isPage(String name) {
+        return Constants.PAGE.equals(name);
+    }
+
+    private boolean isThisAResourceNullCheck(IfStatementTree tree, String ifType) {
+        boolean result = false;
+        if (ifType.equals(tree.condition().kind().name())) {
+            result = isThisANullCheck(tree);
+        }
+        return result;
+    }
+
+    private boolean isRightSideNullCheck(IfStatementTree tree) {
+        return Kind.NULL_LITERAL.equals(tree.condition().lastToken().parent().kind());
+    }
+
+    private boolean isThisANullCheck(IfStatementTree tree) {
```

Comment:
Here's my take on simplifying this method and making it a bit more readable:

```
 private boolean isResourceNullCheck(IfStatementTree tree) {
        SyntaxToken leftOperand = tree.condition().firstToken();
        SyntaxToken rightOperand = tree.condition().lastToken();

        if (leftOperand == null || rightOperand == null) {
            return false;
        }

        return isResourceNullCheck(leftOperand, rightOperand) || isResourceNullCheck(rightOperand, leftOperand);
    }

    private boolean isResourceNullCheck(SyntaxToken operandA, SyntaxToken operandB) {
        return isResourceType(operandA.parent()) && isNullLiteral(operandB.parent());
    }

    private boolean isNullLiteral(Tree operand) {
        return Kind.NULL_LITERAL.equals(operand.kind());
    }

    private boolean isResourceType(Tree operand) {
        return operand instanceof IdentifierTree && Constants.RESOURCE_TYPE.equals(((IdentifierTree)operand).symbolType().fullyQualifiedName());
    }
```

Reply:
Done, looks much better than mine.